Library/Homebrew/formula_specialties.rb

```diff
@@ -18,12 +18,18 @@ def self.url(val)
 # This formula serves as the base class for several very similar
 # formulae for Amazon Web Services related tools.
 class AmazonWebServicesFormula < Formula
+
   # Use this method to peform a standard install for Java-based tools,
   # keeping the .jars out of HOMEBREW_PREFIX/lib
   def install
     rm Dir['bin/*.cmd'] # Remove Windows versions
     libexec.install Dir['*']
-    bin.install_symlink Dir["#{libexec}/bin/*"] - ["#{libexec}/bin/service"]
+    Pathname.glob("#{libexec}/bin/*") do |file|
+      next if file.directory?
+      basename = file.basename
+      next if basename.to_s == "service"
+      (bin/basename).write_env_script file, :JAVA_HOME => ENV["JAVA_HOME"]
```

Comment:
`:JAVA_HOME => ENV["JAVA_HOME"]` is not perfect. It will write the jdk full path into the script. And the full path contains JDK patch version. A idea solution would be `:JAVA_HOME => "$(/usr/libexec/java_home -v <version>)"`. So script won't be break if users' java get update.

The question is how do I implement this? I'm thinking to create something like: `Language::Java.write_env_script`. If so, how do I get the java version tag declared in `depends_on :java => "<version>"`? 

My main idea is to let user use these binary tool without needing to set JAVA_HOME in the `.bashrc/.zshrc`.

Any thought?